### ISSUE TITLE
MEMKIND_ERROR: Persistent hugepages and overcommit hugepages are not available.

### ISSUE THREAD

**Issue by adrianjhpc** [memkind/memkind#706]
Does this error:

```MEMKIND_ERROR: Persistent hugepages and overcommit hugepages are not available.```

Mean that my persistent memory memkind allocations won't actually use memkind?

I have created a memkind_t like this:

```
        int error;
        std::string mount_name = "/mnt/pmem_fsdax";
        std::string pmem_name;
        bool status = eigen_construct_pmem_filename(mount_name, pmem_name);

        // Create data structure to hold the string in charater form to pass to the pmem library
        char pmem_name_char_array[pmem_name.length() + 1];

        // Copy the string into the char array
        strcpy(pmem_name_char_array, pmem_name.c_str());

        error = memkind_create_pmem(pmem_name_char_array, 0, &pmem_kind);
```

But it's not clear when I use `pmem_kind` in `pmem_malloc` it's actually using the persistent memory.

**Comment by bratpiorka** [comment]
This error came from memkind_hbw_check_available() or memkind_hbw_check_available_2mb() functions that are used to check if High Bandwidth memories support huge pages (2MB). You can check this by calling theses functions - more info is available on the memkind_hugetlb(3) man page. But if huge pages are not supported, you are still able to use Memkind to allocate data on Persistent Memory.

I noticed, that you didn't set pmen_name variable in your code. Besides this, memkind_create_pmem accepts as a first argument a path to a directory where the PMEM is mounted, not a path to a file.

About pmem_malloc example - could you rephrase your problem? I'm not sure how I can help you.

**Comment by adrianjhpc** [comment]
`pmem_name` is set by the function `eigen_construct_pmem_filename` (I've not included it for brevity) which takes the `mount_name` and adds the socket number ensure locality is respected. `pmem_name_char_array` is the directory where the PMEM is mounted.

I was just checking that with the warning above `memkind_malloc` isn't defaulting to normal memory instead of persistent memory, but it sounds like from what you're saying that isn't the case.

**Comment by bratpiorka** [comment]
@adrianjhpc - is there anything else I can do for you or can I close this issue?

**Comment by adrianjhpc** [comment]
No, this can be closed.
